vertx-template-engines/vertx-web-templ-freemarker/src/main/java/io/vertx/ext/web/templ/impl/FreeMarkerTemplateEngineImpl.java

```diff
@@ -62,8 +63,9 @@ public FreeMarkerTemplateEngine setMaxCacheSize(int maxCacheSize) {
   }
 
   @Override
-  public void render(RoutingContext context, String templateFileName, Handler<AsyncResult<Buffer>> handler) {
+  public void render(RoutingContext context, String templateDirectory, String templateFileName, Handler<AsyncResult<Buffer>> handler) {
     try {
+      templateFileName = templateDirectory + templateFileName;
```

Comment:
what if `templateDirectory` is `null` or `""` then you either get messed up paths or you just allow looking up from the root of the filesystem. This could be a security issue, I believe we should apply the path normalize algorithm already used on vert.x web